app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from "react";
+
+import { Formik, Form } from "formik";
+import { Pane, Toastr, Button } from "neetoui";
+import { Input, Select } from "neetoui/formik";
+
+import {
+  CONTACTS_FORM_VALIDATION_SCHEMA,
+  ROLE_LIST,
+  CONTACTS_FORM_INITIAL_FORM_VALUES,
+} from "../constants";
+
+const ContactForm = ({ onClose }) => {
+  const [submitted, setSubmitted] = useState(false);
+
+  const handleSubmit = () => {
+    setSubmitted(true);
+    Toastr.success("New contact added");
+    onClose();
+  };
+
+  return (
+    <Formik
+      initialValues={CONTACTS_FORM_INITIAL_FORM_VALUES}
+      validateOnBlur={submitted}
+      validateOnChange={submitted}
+      validationSchema={CONTACTS_FORM_VALIDATION_SCHEMA}
+      onSubmit={handleSubmit}
+    >
+      {({ isSubmitting }) => (
+        <Form className="w-full">
+          <Pane.Body className="space-y-8">
+            <div className="flex w-full flex-row space-x-6">
+              <Input
+                required
+                className="w-full flex-grow-0"
+                label="First Name"
+                name="firstname"
+                placeholder="Enter first name"
+              />
+              <Input
+                required
+                className="w-full flex-grow-0"
+                label="Last Name"
+                name="lastname"
+                placeholder="Enter last name"
+              />
+            </div>
+            <Input
+              required
+              className="w-full flex-grow-0"
+              label="Email Address"
+              name="email"
+              placeholder="Enter your email address"
+            />
+            <Select
+              isSearchable
+              required
+              className="w-full flex-grow-0"
+              label="Role"
+              name="role"
+              options={ROLE_LIST}
+              placeholder="Select Role"
+            />
+          </Pane.Body>
+          <Pane.Footer>
+            <Button
+              className="mr-3"
+              disabled={isSubmitting}
+              label="Save Changes"
+              loading={isSubmitting}
+              size="large"
+              style="primary"
+              type="submit"
+            />
+            <Button
+              label="Cancel"
+              size="large"
+              style="text"
+              onClick={onClose}
+            />
```

Comment:
Go through the Slack thread.

Reply:
I had a small doubt regarding this. 
By adding `type="reset"`, I can remove the `onClick` listener right? So, in order to close the pane, the user will have to click away from the Pane, or the "X" mark on the top right corner.

Reply:
@aagneyanNamboodiri `onClick` should be kept intact. That would have some side effects. You can also pass in the `type=reset`. This is to make things explicit that this button will `reset` the values upon executing.

>So, from a UI perspective isn't it better if the button is labelled as "Reset" or "Reset values" rather than "Cancel"

No. Here we are cancelling an action, which is either delete or add etc. `Reset` is a technical term that should only be used when we don't have any other alternatives.

Reply:
Understood. 
I will add this with any changes during the final review.